### PR TITLE
Replacement of Init-Container Specific Tests

### PR DESCRIPTION
**Pull request by ifilonenko**
With the removal of the init-container the tests need to refactored. For the sake of testing pure jar/file distribution it might make more sense to add tests into Spark Core Examples directly. 

For now, this could satisfy just the verification that spark-files is being distributed properly by spark-submit.

### CODE REVIEW

File: src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala [hunk 1]
```diff
@@ -170,41 +170,18 @@ private[spark] class KubernetesSuite extends FunSuite with BeforeAndAfterAll wit
...
+  test("Test extraJVMProprties being present on Driver") {
     sparkAppConf
-      .set("spark.kubernetes.mountDependencies.filesDownloadDir",
-        CONTAINER_LOCAL_FILE_DOWNLOAD_PATH)
-      .set("spark.files", REMOTE_PAGE_RANK_DATA_FILE)
-    runSparkPageRankAndVerifyCompletion(
-      appArgs = Array(CONTAINER_LOCAL_DOWNLOADED_PAGE_RANK_DATA_FILE))
+      .set("spark.driver.extraJavaOptions", "-Dspark.test.foo=spark.test.bar")
+    runSparkJVMCheckAndVerifyCompletion(
+      expectedJVMValue=Seq("spark.test.foo","spark.test.foo"))
```
Comment (mccheah) [inline src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala]: Think we probably want something more specific, e.g. "The property spark.test.foo was properly set to spark.test.bar"

Comment (ifilonenko) [reply src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala]: The result in the test is printing as a tuple (spark.test.foo,spark.test.bar) to show k-v pair. Will that suffice?

**Comment by ifilonenko** [comment]
This needs to be merged in but is currently failing to build because of the non-updated spark-distribution that is being used to test integration test success @ssuchter 

Otherwise this is ready to push, and should be pushed for future PRs into Spark.

**Review by liyinan926 — APPROVED** [review]
The changes look good to me.

**Comment by ifilonenko** [comment]
Can I merge?

**Comment by ssuchter** [comment]
I set up a new PRB to test changes to the integration tests. It passed, as you can see. Is this the validation you were looking for?

I notice you said this:

> This needs to be merged in but is currently failing to build because of the non-updated spark-distribution that is being used to test integration test success

Is it surprising this passed?

**Comment by ifilonenko** [comment]
Nope, this was supposed to pass. All good. Can someone merge please

**Review by mccheah — APPROVED** [review]

**Pull request merged**